src/C++/SSLSocketConnection.h

```diff
@@ -164,17 +164,32 @@ class SSLSocketConnection : Responder
       m_pMonitor->signal( m_socket );
   }
 
+  void subscribeToSocketWriteAvailableEvents() {
+      m_pMonitor->signal(m_socket);
+  }
```

Comment:
This is an alternative to signal, without the check if the queue has data. It is intended to be used from readFromSocket() when it wants to write due to a re-negotiation of ssl keys.